### PR TITLE
Search results configurable model

### PR DESCRIPTION
**Pull request by yerramshilpa**
## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [ ] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [ ] Internet Explorer 11
- [ ] Edge
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

## Checklist:

- [ ] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ ] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/packages/components/src/lib/search-result-list/search-result-list.component.html [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
-      <div class="grid-col-fill display-flex flex-column padding-y-1">
-        <h1 class="margin-bottom-105 text-semibold">No matches found</h1>
-        <p class="margin-y-0 font-sans-lg">We couldn't find a match for your search criteria.</p>
-        <p class="margin-y-0 font-sans-lg">Please try another search or go back to previous results.</p>
-        <button class="usa-button usa-button--secondary width-card margin-y-2">Go Back</button>
+    </div>
+  </div>
+</ng-container>
+<ng-container *ngIf="updateModel?.metadata?.messages" class="sds-card--shadowed sds-card--bordered margin-bottom-2">
+  <div *ngFor="let msg of updateModel.metadata.messages" class="sds-card__content grid-row" [ngClass]="msg.classes ? msg.classes : 'border-1px border-base-lighter bg-base-lightest '">
```
Comment (davereed) [inline libs/packages/components/src/lib/search-result-list/search-result-list.component.html:25]: We should leave the card classes for the base styling here, just allow them to add the additional classes they want. If they need more customization we will be adding template customization in the future.

Also I think we should move the additional classes to the top level for overall targeting.

File: libs/packages/components/src/lib/search-result-list/search-result-list.component.html [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
       </div>
-      <div class="grid-col-fill display-flex flex-column padding-y-1">
-        <h1 class="margin-bottom-105 text-semibold">No matches found</h1>
-        <p class="margin-y-0 font-sans-lg">We couldn't find a match for your search criteria.</p>
-        <p class="margin-y-0 font-sans-lg">Please try another search or go back to previous results.</p>
-        <button class="usa-button usa-button--secondary width-card margin-y-2">Go Back</button>
+    </div>
+  </div>
+</ng-container>
+<ng-container *ngIf="updateModel?.metadata?.messages" class="sds-card--shadowed sds-card--bordered margin-bottom-2">
```
Comment (davereed) [inline libs/packages/components/src/lib/search-result-list/search-result-list.component.html:24]: This should be the card body styles, where is sds-card?

File: libs/packages/components/src/lib/search-result-list/search-result-list.component.html [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
+  <div *ngFor="let msg of updateModel.metadata.messages" class="sds-card__content grid-row" [ngClass]="msg.classes ? msg.classes : 'border-1px border-base-lighter bg-base-lightest '">
+    <div class="grid-col-auto text-center text-base margin-4">
+      <fa-icon [icon]="(msg.type ==='error') ?  ['sds', 'alert-error'] : ['sds', 'information-circle']" [classes]="['search']"
+        size="5x"></fa-icon>
+    </div>
+    <div class="grid-col-fill display-flex flex-column padding-y-1">
+      <h1 class="margin-bottom-105 text-semibold">{{msg.title}}</h1>
+      <p class="margin-y-0 font-sans-lg" [innerHTML]="msg.message"></p>
+      <div class="margin-top-3">
+        <button *ngFor="let button of msg.buttons" class="usa-button  width-card" [ngClass]="button.classes ? button.classes : 'usa-button--secondary'"
```
Comment (davereed) [inline libs/packages/components/src/lib/search-result-list/search-result-list.component.html:34]: Buttons should always have an action hooked up, we consider requiring it in the interface or omitting it on the template if it doesn't do anything.

Comment (yerramshilpa) [reply libs/packages/components/src/lib/search-result-list/search-result-list.component.html]: Updated.

File: libs/packages/components/src/lib/search-result-list/search-result-list.component.ts [hunk 4]
```diff
@@ -1,23 +1,45 @@
...
   templateUrl: './search-result-list.component.html',
   styleUrls: ['./search-result-list.component.scss']
 })
-export class SdsSearchResultListComponent {
 
-  /**
-   * List of items
+export class SdsSearchResultListComponent  {
+
+  public updateModel = new SearchModel();;
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/search-result-list/search-result-list.component.ts]: removed

File: libs/packages/components/src/lib/search-result-list/search-result-list.component.html (hunk 1 above)
Comment (yerramshilpa) [reply libs/packages/components/src/lib/search-result-list/search-result-list.component.html]: Updated with new css

**Pull request merged**